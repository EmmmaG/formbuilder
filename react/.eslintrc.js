module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
}
