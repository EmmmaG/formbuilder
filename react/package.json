{
  "name": "hcard-builder",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.2",
    "@types/react-dom": "^16.8.3",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "2.1.8",
    "typescript": "^3.3.4000"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "description": "This project was to conver a fixed format file into csv",
  "main": "index.js",
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",
    "@types/enzyme": "^3.1.11",
    "@types/enzyme-adapter-react-16": "^1.0.1",
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.25.0",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^24.1.0",
    "jest-enzyme": "^7.0.1",
    "jest-fetch-mock": "^1.6.5",
    "ts-jest": "^22.4.6"
  },
  "author": "",
  "license": "ISC"
}
