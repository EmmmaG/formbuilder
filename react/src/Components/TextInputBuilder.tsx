import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'


const styles = () => ({
    title: {
        fontSize: '25px',
        margin: '10px',
    },
    fieldContainer: {
        margin: '25px',
    }
})

export interface TextInput {
    id?: number
    type: 'textInput'
    question: string
    placeholder?: string
    defaultValue?: string
    label?: string
}

type TextInputKey = keyof TextInput

export interface Props extends WithStyles<typeof styles> {
    onCreate: (textInputValue: TextInput) => void
}

export interface State {
    textInputValue: TextInput
}

const emptyTextInput = {
    type: 'textInput' as 'textInput',
    question: '',
    placeholder: '',
    defaultValue: '',
    label: '',
}

export class TextInputBuilder extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            textInputValue: emptyTextInput
        }
    }

    private onTextFieldChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>, textInputKey: TextInputKey) => {
        const currentTextInputValue = this.state.textInputValue
        const newValue = e.target.value

        this.setState({
            textInputValue: {
                ...currentTextInputValue,
                [textInputKey]: newValue,
            }
        })
    }

    private onCreate = () => {
        const { textInputValue } = this.state

        this.props.onCreate(textInputValue)

        this.setState({textInputValue: emptyTextInput})
    }

	public render() {
        const { classes } = this.props
        const { textInputValue } = this.state

		return (
			<div>
                <Typography className={classes.title}>Create a text input</Typography>

                <div className={classes.fieldContainer}>
                    <Typography>* Please add a question for your text input (Must Have)</Typography>
                    <TextField
                        id='question'
                        label="Text input question"
                        value={textInputValue.question}
                        onChange={(e) => this.onTextFieldChange(e, 'question')}
                        margin='normal'
                    />
                </div>

                <div className={classes.fieldContainer}>
                    <Typography>Please add a placeholder for your text input (Optional)</Typography>
                    <TextField
                        id='placeholder'
                        label="Text input placeholder"
                        value={textInputValue.placeholder}
                        onChange={(e) => this.onTextFieldChange(e, 'placeholder')}
                        margin='normal'
                    />
                </div>

                <div className={classes.fieldContainer}>
                    <Typography>Please add a display label for your text input (Optional) </Typography>
                    <TextField
                        id='label'
                        label="Text input label"
                        value={textInputValue.label}
                        onChange={(e) => this.onTextFieldChange(e, 'label')}
                        margin='normal'
                    />
                </div>

                <div className={classes.fieldContainer}>
                    <Typography>Please add a default text for your text input (Optional)</Typography>
                    <TextField
                        id='defaultValue'
                        label="Text input defaultValue"
                        value={textInputValue.defaultValue}
                        onChange={(e) => this.onTextFieldChange(e, 'defaultValue')}
                        margin='normal'
                    />
                </div>

                <Button
                    variant='outlined'
                    color='primary'
                    className={classes.fieldContainer}
                    onClick={this.onCreate}
                    disabled={!textInputValue.question}
                >
                    Create Text Input
                </Button>
            </div>
		)
	}
}

export default withStyles(styles)(TextInputBuilder)
