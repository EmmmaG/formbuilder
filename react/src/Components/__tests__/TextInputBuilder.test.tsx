import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'
import TextField from '@material-ui/core/TextField'

import TextInputBuilder, { Props } from '../TextInputBuilder'

const props: Props = {
    onCreate: jest.fn(),
	classes: idObj,
}

describe('TextInputBuilder', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<TextInputBuilder {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})

	it('sets correct value', () => {
		const wrapper = shallow(<TextInputBuilder {...props} />).dive()
		const questionField = wrapper.find(TextField).findWhere(t => t.prop('id') === 'question')

		questionField.prop('onChange')({ target: { value: 'weather?' } })

		expect(wrapper.state('textInputValue')).toEqual(expect.objectContaining({
			question: 'weather?',
		}))
	})
})
