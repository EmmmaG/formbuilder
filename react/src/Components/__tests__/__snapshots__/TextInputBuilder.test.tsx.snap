// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputBuilder renders correctly 1`] = `
<div>
  <WithStyles(Typography)
    className="TextInputBuilder-title-1"
  >
    Create a text input
  </WithStyles(Typography)>
  <div
    className="TextInputBuilder-fieldContainer-2"
  >
    <WithStyles(Typography)>
      * Please add a question for your text input (Must Have)
    </WithStyles(Typography)>
    <TextField
      id="question"
      label="Text input question"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
  </div>
  <div
    className="TextInputBuilder-fieldContainer-2"
  >
    <WithStyles(Typography)>
      Please add a placeholder for your text input (Optional)
    </WithStyles(Typography)>
    <TextField
      id="placeholder"
      label="Text input placeholder"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
  </div>
  <div
    className="TextInputBuilder-fieldContainer-2"
  >
    <WithStyles(Typography)>
      Please add a display label for your text input (Optional) 
    </WithStyles(Typography)>
    <TextField
      id="label"
      label="Text input label"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
  </div>
  <div
    className="TextInputBuilder-fieldContainer-2"
  >
    <WithStyles(Typography)>
      Please add a default text for your text input (Optional)
    </WithStyles(Typography)>
    <TextField
      id="defaultValue"
      label="Text input defaultValue"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
  </div>
  <WithStyles(Button)
    className="TextInputBuilder-fieldContainer-2"
    color="primary"
    disabled={true}
    onClick={[Function]}
    variant="outlined"
  >
    Create Text Input
  </WithStyles(Button)>
</div>
`;
